event_lead_event/models/event.py

```diff
@@ -0,0 +1,27 @@
+# Copyright 2018 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import _
+from odoo import api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class Event(models.Model):
+
+    _inherit = "event.event"
+
+    lead_event = fields.Boolean(
+        string='Lead Event',
+        index=True,
+        default=False,
+    )
+
+    @api.constrains('lead_event')
+    def check_only_one_leadevent(self):
+        """Make sure we only ever have one lead event at a time."""
+        if self.search_count([('lead_event', '=', True)]) > 1:
```

Comment:
does this work correctly when creating a second event?
there is only one database record at that moment :)

Reply:
`> 1` -> how could that change the behavior?

Reply:
Tested with @tschanzt, as this function is called after the update in the same transaction, it works.

Reply:
I wasn't sure about when that constraint is checked and if that one respects a fresh record that has been just created :smile: